pandasai/smart_datalake/__init__.py

```diff
@@ -648,34 +473,68 @@ def _get_output_type_hint(self, output_type: Optional[str]) -> str:
                 of the above or "text".
 
         Returns:
-            (str): Output type hint
+            PipelineContext: The Pipeline Context to be used by Smart Data Lake Pipeline.
         """
 
-        output_type_helper = output_type_factory(output_type, logger=self.logger)
-        return output_type_helper.template_hint
+        self._query_exec_tracker.start_new_track()
 
-    def _get_viz_library_type(self) -> str:
-        """
-        Get the visualization library type based on the configured library.
+        self.logger.log(f"Question: {query}")
+        self.logger.log(f"Running PandasAI with {self._llm.type} LLM...")
 
-        Returns:
-            (str): Visualization library type
-        """
+        self._assign_prompt_id()
+
+        self._query_exec_tracker.add_query_info(
+            self._conversation_id, self._instance, query, output_type
+        )
+
+        self._query_exec_tracker.add_dataframes(self._dfs)
+
+        self._memory.add(query, True)
 
+        output_type_helper = output_type_factory(output_type, logger=self.logger)
         viz_lib_helper = viz_lib_type_factory(self._viz_lib, logger=self.logger)
-        return viz_lib_helper.template_hint
 
-    def _add_result_to_memory(self, result: dict):
+        pipeline_context = PipelineContext(
+            dfs=self.dfs,
+            config=self.config,
+            memory=self.memory,
+            cache=self.cache,
+            query_exec_tracker=self._query_exec_tracker,
+        )
+        pipeline_context.add_intermediate_value("is_present_in_cache", False)
+        pipeline_context.add_intermediate_value(
+            "output_type_helper", output_type_helper
+        )
+        pipeline_context.add_intermediate_value("viz_lib_helper", viz_lib_helper)
+        pipeline_context.add_intermediate_value(
+            "last_code_generated", self._last_code_generated
+        )
+        pipeline_context.add_intermediate_value("get_prompt", self._get_prompt)
+        pipeline_context.add_intermediate_value("last_prompt_id", self.last_prompt_id)
```

Comment:
last prompt id i think it can be added in last prompt id. We need to store thing in right place and i would let's completely remove last_prompt_id from SmartDatalake and initialize in QueryTracker -> start_new_track method.

Reply:
done